ta4j-core/src/main/java/org/ta4j/core/num/BigDecimalNum.java

```diff
@@ -181,6 +181,36 @@ public Num pow(int n) {
         return new BigDecimalNum(delegate.pow(n, MATH_CONTEXT));
     }
 
+    /**
+     * Returns a {@code Num} whose value is <tt>(this<sup>n</sup>)</tt>.
+     * @param n power to raise this {@code Num} to.
+     * @return <tt>this<sup>n</sup></tt>
+     * @see BigDecimal#pow(int, java.math.MathContext)
+     */
+    public Num pow(Num n) {
+        // Because BigDecimal.pow(BigDecimal) is not supported we need to find another way, we could do:
+        // Math.pow(a.doubleValue(), b.doubleValue()), but there is a better way:
+        // Perform X^(A+B)=X^A*X^B (B = remainder)
+        // As suggested: https://stackoverflow.com/a/3590314
+        BigDecimal result;
+        BigDecimal power = ((BigDecimalNum)n).delegate;
```

Comment:
Just double checked and this may actually be consistent with all of the other operations.  It just seems wrong to me that you can't
```
Num a = DoubleNum.valueOf("1.0");
Num b = BigDecimalNum.valueOf("2.0");
assertNumEquals("3.0", a.plus(b));
```
But I'm willing to let this go for now.

Reply:
> Any Num should be able to be raised to the pow of any other Num regardless of implementation.
I do not agree with this assumption. `Num#pow` could be overloaded with int/double, but different `Num` implementation should not be compatible in any way.

Reply:
Okay, I'll let it go.  :)  I now approve of line 196.